code-reviews.md

```diff
@@ -0,0 +1,74 @@
+# Code reviews at FUN
+
+Code reviews are collaborative steps in software development.
+The purpose of code reviews is twofold. They allow both to:
+* check the quality of the code and
+* to communicate actively on the evolution of the software.
+
+Code reviews concern all members who are involved in software development, regardless of their competence level.
+This concept raises questions about their effectiveness and relevance both on the side of the developer submitting the review and on the side of the developers reviewing the code.
+
+> When ? How? How long do we review codes?
+> Which focus? Which feedback to give to the reviewer?
+
+## Approach of code reviews
+
+Repository managers such as GitHub or GitLab are helpful to review codes.
+Associated workflows, pull request \(PR\) or merge request \(MR\) ease the tracking of information shared in code reviews.
+
+We recommend considering a fine granularity of code reviews.
+The benefit is both temporal and qualitative.
+By doing code reviews as frequently as possible, the exercise is made easier and all members feel more involved in development.
+By doing light and targeted code reviews on a specific topic, the review is done with more attention because it is less complex.
+
+A code review is an opportunity to **actively** communicate about the changes to other developers and get feedback on the quality of the changes.
+As far as possible, a code review should be done for small changes.
+In this perspective, we advise against doing code reviews for a collection of changes that can be analyzed independently.
+However, it is important to keep in mind that what is under review must be understandable in its software context.
+Complex refactorings or features can not be splitted in small changes as they would lose software design information.
+The review would then bypass the algorithmic choices in favor of a simple technical review.
+
+When starting a review, we recommend to assess the granularity of the review to keep visible the algorithm and design of the feature.

```

Comment:
It seems that this advice is targeted to the developer, not the reviewer?

Reply:
Yes it is. It is for good practices of reviews, all along the process. Developers are also responsible to start the review on good basis. A reviewer can notice that the review to be done isn't well sructured and inform the developer.

Reply:
There's something wrong in the sentence I think. Should it read "when opening a Pull Request (vs MR)?

Reply:
Tell me if I am wrong, I consider opening a pull request as the first step/start of a review

Reply:
I would say the developer opens a PR and the reviewer starts reviewing (hence starts the review). So it's not the developer who starts a review 😉

Reply:
Ok, let's write it like this! I'll make the modification